src/test/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtServiceTest.java

```diff
@@ -111,30 +115,47 @@ void shouldThrowRIllegalArgumentExceptionOnUnknownCourtId() {
     @Test
     void shouldReturnHearingCourt() {
         PilotCourtService pilotCourtService = new PilotCourtService(
-            CSV_PATH_SINGLE,
+            CSV_PATH_COURT_IDS,
             courtFinderApi,
             hearingCourtMapper,
             appInsights
         );
 
-        CourtFinderResponse courtFinderResponse = CourtFinderResponse.builder().build();
+        CourtFinderResponse courtFinderResponse = DataFactory.createCourtFinderResponseFromJson(COURT_FINDER_RESPONSE_NEWCASTLE);
+
+        Request request = Request.create(Request.HttpMethod.GET, "URL", ImmutableMap.of(), Request.Body.empty(), null);
+        when(courtFinderApi.findMoneyClaimCourtByPostcode(anyString())).thenThrow(FeignException.errorStatus("",
+                Response.builder().request(request).build()))
+            .thenReturn(courtFinderResponse);
+
+        pilotCourtService.init();
+
+        HearingCourt expectedHearingCourt = DataFactory.createHearingCourtFromJson(HEARING_COURT_NEWCASTLE);
+
         when(courtFinderApi.findMoneyClaimCourtByPostcode(anyString()))
             .thenReturn(courtFinderResponse);
 
-        HearingCourt hearingCourt = HearingCourt.builder().name("SAMPLE COURT").build();
-        when(hearingCourtMapper.from(any(Court.class))).thenReturn(hearingCourt);
+        // TODO :
+        Court oldCourt = Court.builder().slug("newcastle-civil-family-courts-and-tribunals-centre").build();
 
-        pilotCourtService.init();
+        when(new CourtFinderContainer(courtFinderApi).getCourtFromCourtFinderResponse(any()))
+            .thenReturn(oldCourt);
+
+        CourtDetails courtDetails = DataFactory.createCourtDetailsFromJson(COURT_DETAILS_NEWCASTLE);
+
+        when(courtFinderApi.getCourtDetailsFromNameSlug(anyString()))
+            .thenReturn(courtDetails);
 
         HearingCourt actualHearingCourt = new HearingCourt();
-        Optional<HearingCourt> birminghamHearingCourt = pilotCourtService.getPilotHearingCourt("BIRMINGHAM");
-        if (birminghamHearingCourt.isPresent()) {
-            actualHearingCourt = birminghamHearingCourt.get();
+        Optional<HearingCourt> hearingCourt = pilotCourtService.getPilotHearingCourt("NEWCASTLE");
+        if (hearingCourt.isPresent()) {
+            actualHearingCourt = hearingCourt.get();
         }
```

Comment:
`actualHearingCourt = hearingCourt.orElseGet(HearingCourt::new)` would it work here?